### PR TITLE
ADD: setting to redact push notification content

### LINKED ISSUE #8499: Privacy mode for push notifications
Add a setting to GroundControl that all PN texts should be redacted (blank or generic). Data payload should not be transmitted as well in that case.
BW should display that setting in settings

(to prevent apple/google from reading your financial info)

### PR DESCRIPTION
**Pull request by r6mez**
Adds a "Redact notification content" toggle to Notification Settings. When enabled, GroundControl sends generic push text with no transaction data, so payment details never appear on the lock screen or in the notification payload.

depends on https://github.com/BlueWallet/GroundControl/pull/329 
closes https://github.com/BlueWallet/BlueWallet/issues/8499

Screenshot_1782376142

### CODE REVIEW

File: loc/en.json [hunk 1]
```diff
@@ -329,6 +329,8 @@
   "notifications": {
     "would_you_like_to_receive_notifications": "Would you like to receive notifications when you get incoming payments?",
     "notifications_subtitle": "Incoming payments and transaction confirmations",
+    "redact_notifications": "Redact notification content",
+    "redact_notifications_subtitle": "Hide payment details in notifications.",
```
Comment (limpbrains) [inline loc/en.json]: Subtitle is actually more informative that the title here
I suggest changing title to "Hide payment details"
And subtitle then to something like
"Notifications won't show amounts or transaction info on your lock screen"
or 
"Receive generic alerts without payment details"
or 
"Keeps amounts and sender info off your lock screen"

**Review by limpbrains — APPROVED** [review]

File: loc/en.json (hunk 1 above)
Comment (r6mez) [reply loc/en.json]: Changed title to "Hide payment details" and subtitle to "Notifications won't show amounts or transaction info on your lock screen." Thanks for your review.

File: blue_modules/notifications.ts [hunk 2]
```diff
@@ -460,6 +460,33 @@ export const setLevels = async (levelAll: boolean) => {
...
+ */
+export const setRedactNotifications = async (redacted: boolean) => {
+  const pushToken = await getPushToken();
+  if (!pushToken?.token || !pushToken?.os) return;
+
+  const response = await fetch(`${baseURI}/setTokenConfiguration`, {
+    method: 'POST',
+    headers: _getHeaders(),
+    body: JSON.stringify({ redacted, token: pushToken.token, os: pushToken.os }),
+  });
```
[the remark this answers is not in the document]
Comment (r6mez) [reply blue_modules/notifications.ts:480]: fixed

File: screen/settings/NotificationSettings.tsx [hunk 3]
```diff
@@ -135,6 +149,8 @@ const NotificationSettings: React.FC = () => {
           await updateNotificationStatus();
         }
 
+        setRedactedState(await isNotificationsRedacted());
```
[the remark this answers is not in the document]
Comment (r6mez) [reply screen/settings/NotificationSettings.tsx]: fixed

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: blue_modules/notifications.ts [hunk 4]
```diff
@@ -460,6 +460,35 @@ export const setLevels = async (levelAll: boolean) => {
...
 };
 
+/**
+ * Posts to groundcontrol whether push notification text/data
+ *  for this device should be redacted
+ *
+ * @param redacted {Boolean}
+ * @returns {Promise<void>}
+ */
+export const setRedactNotifications = async (redacted: boolean) => {
```
Comment (GladosBlueWallet) [inline blue_modules/notifications.ts:470]: **[MEDIUM]** `setRedactNotifications` flips a server flag and stops. `setLevels(false)` at 448 clears delivered notifications; this path does not. Enable redaction and yesterday's '+1000 sats' banners keep waving at Apple. Stunning privacy theater.

File: blue_modules/notifications.ts [hunk 5]
```diff
@@ -460,6 +460,35 @@ export const setLevels = async (levelAll: boolean) => {
...
+    headers: _getHeaders(),
+    body: JSON.stringify({ redacted, token: pushToken.token, os: pushToken.os }),
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to set redact configuration: ' + response.statusText);
+  }
+};
+
+export const isNotificationsRedacted = async (): Promise<boolean> => {
```
Comment (GladosBlueWallet) [inline blue_modules/notifications.ts:487]: **[MEDIUM]** `isNotificationsRedacted` trusts `getLevels()`, which swallows errors into `{}` at 656–658. Network hiccup renders the toggle OFF while the server stays redacted — or the reverse. Misleading UI on a privacy control. I do love unreliable experiments.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Privacy feature for GroundControl pushes — toggle, API wiring, strings. GroundControl strips payloads as designed; this client never adapted. Android test subjects lose silent payment sync. iOS limps via a delivered-notification kludge. Ark locals and old tray entries still leak. Zero tests. The cake is a lie; so is full privacy.

### Inline findings (could not anchor on diff)

- `hooks/useCompanionListeners.ts:163` — **[HIGH]** GroundControl redaction deletes `type`/`address`/`txid`. Your wallet lookup NOPs. iOS cheats with `refreshAllWalletTransactions` at 263–264; Android's `getDeliveredNotifications` is hardcoded `[]` at notifications.ts:722–724. Half your subjects get a privacy toggle that silently stops payment sync. Congratulations.
- `blue_modules/arkade-notifications.ts:124` — **[MEDIUM]** Issue #8499 demanded *all* push text redacted. `notifyArkSwapActionable` never reads `isNotificationsRedacted` and still posts wallet labels to the lock screen. GroundControl got the blindfold; Ark swaps did not. Science!
- `hooks/useCompanionListeners.ts:144` — **[MEDIUM]** Tap a redacted push — generic body, zero routing fields — and wallet lookup fails with no fallback navigation. Your test subject pokes 'You have a new notification' and nothing happens. Delightful UX for the enrichment center.
- `screen/settings/NotificationSettings.tsx:174` — **[MEDIUM]** AppState resume refreshes notification enabled state but never re-fetches redaction. Stale toggle vs server. CONTRIBUTING.md line 29 also wants tests; this PR added none for `setRedactNotifications`, `isNotificationsRedacted`, or the switch. Even I write test protocols.

**Comment by Overtorment** [comment]
GC deployed

**Comment by Overtorment** [comment]
tested this end-to-end with fresh deployed server, worked as expected
